insights/core/plugins.py

```diff
@@ -190,6 +192,28 @@ def register_cluster_reducer(func):
     p["module"] = func.__module__
 
 
+def deprecated(reason='', pending=True):
```

Comment:
My gut tells me that `pending=False` will be more common.

Reply:
Yeah, let's just use DeprecationException throughout.  It was a silly idea.

Reply:
We should probably require a `reason` to be set with some information on what the user should use instead.  Requiring that parameter will simplify the message later in this function as well.

Reply:
Good point - I'll call it 'solution' instead of 'reason'.